### PR TITLE
Fix metabox (apt profile + daemon respawn) (BugFix)

### PR DESCRIPTION
**Pull request by kissiel**
## Description
efe766555be19c59ba8d495257739996cc95065e
The previous "on-failure" meant that sending a signal to agent (for instance SIGKILL) would completely terminate the agent and make systemd not restart it.
This is very bad for emulating agent crashes. Thus I'm proposing here to `always` do it.

50e3f4337e665627a0d2b0c0b16017f1575b1a21 and 33f5ecce941dbd2eb9cfff681f3300ffefc851c8
Make agents from snaps **always** respawn
    
Considering that Checkbox is a tool that's intended to be sometimes killed, it needs to always be brought back online (its agent).
Immediately this fixes the agent respawn metabox tests and goes in line with the changes being made to the auto agent respawn stories.

ec27e416d399221df84c273a3a4b49d34b2c903c

Remove apt-related pre-provisioning from lxd profiles in Metabox
    
This is an relic from the times when we wanted to improve container provisioning speed, but now, when testing from source we provision everything with a normal apt execution, so this can go away. Also this list had a potential to quickly go stale.

## Testing
Tested with MB runs using an agent-killing scenario.

### CODE REVIEW

**Review by Hook25 — APPROVED** [review]
+1, agree

EDIT: makes me wonder, should we also change the deb and snap? they have the same strategy

**Review by Hook25 — CHANGES_REQUESTED** [review]
If you want to change them all here, I'm fine with it but:
- change the title of the PR
- change also the deb service

**Review by yphus — CHANGES_REQUESTED** [review]
The service file provided by the checkbox-ng deb package also needs the same fix:
https://github.com/canonical/checkbox/blob/main/checkbox-ng/debian/checkbox-ng.service

File: checkbox-snap/series_classic16/snap/snapcraft.yaml [hunk 1]
```diff
@@ -28,7 +28,7 @@ apps:
     command-chain: [bin/wrapper_local]
     command: bin/checkbox-cli-wrapper run-agent
     daemon: simple
-    restart-condition: on-failure
+    restart-condition: always
```
[the remark this answers is not in the document]
Comment (kissiel) [reply checkbox-snap/series_classic16/snap/snapcraft.yaml:31]: Maybe let's do that even if only for consistency :)

File: metabox/metabox/lxd_profiles/checkbox.profile
[the remark this answers is not in the document]
Comment (kissiel) [reply metabox/metabox/lxd_profiles/checkbox.profile:1]: This is explained in the PR description.

**Comment by kissiel** [comment]
> The service file provided by the checkbox-ng deb package also needs the same fix: https://github.com/canonical/checkbox/blob/main/checkbox-ng/debian/checkbox-ng.service

Updated in a3c509e40ad54be855bb4bd591ce911b713c233e

**Review by Hook25 — APPROVED** [review]
+1, ty for updating them all!

**Review by yphus — APPROVED** [review]
LGTM

**Review by Hook25 — DISMISSED** [review]
+1, ty

**Review by yphus — DISMISSED** [review]
+1

**Comment by kissiel** [comment]
I also removed the glxgears test that was silently failing before, but now it started failing properly and we really don't want to have it tested.

**Review by Hook25 — APPROVED** [review]
+1, ty

**Pull request merged**